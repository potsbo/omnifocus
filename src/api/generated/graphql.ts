import { GraphQLResolveInfo } from 'graphql';
import { GraphQLClient } from 'graphql-request';
import * as Dom from 'graphql-request/dist/types.dom';
import gql from 'graphql-tag';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type RequireFields<T, K extends keyof T> = Omit<T, K> & { [P in K]-?: NonNullable<T[P]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Project = {
  __typename?: 'Project';
  id: Scalars['String'];
  name: Scalars['String'];
  rootTask: Task;
};

export type Query = {
  __typename?: 'Query';
  flattenedTasks: Array<Task>;
  inboxTasks: Array<Task>;
  project?: Maybe<Project>;
};


export type QueryFlattenedTasksArgs = {
  available?: InputMaybe<Scalars['Boolean']>;
  flagged?: InputMaybe<Scalars['Boolean']>;
  limit?: InputMaybe<Scalars['Int']>;
  withEffectiveDueDate?: InputMaybe<Scalars['Boolean']>;
};


export type QueryInboxTasksArgs = {
  available?: InputMaybe<Scalars['Boolean']>;
  flagged?: InputMaybe<Scalars['Boolean']>;
};


export type QueryProjectArgs = {
  id: Scalars['String'];
};

export type Task = {
  __typename?: 'Task';
  completed: Scalars['Boolean'];
  containingProject?: Maybe<Project>;
  effectiveDueDate?: Maybe<Scalars['String']>;
  effectivelyCompleted: Scalars['Boolean'];
  flagged: Scalars['Boolean'];
  id: Scalars['String'];
  name: Scalars['String'];
  tasks?: Maybe<Array<Task>>;
};

export type TaskViewModelFragment = { __typename?: 'Task', name: string, id: string, effectiveDueDate?: string | null, completed: boolean, effectivelyCompleted: boolean, flagged: boolean, containingProject?: { __typename?: 'Project', id: string, name: string } | null };

export type GetTasksQueryVariables = Exact<{
  flagged?: InputMaybe<Scalars['Boolean']>;
  available?: InputMaybe<Scalars['Boolean']>;
  withEffectiveDueDate?: InputMaybe<Scalars['Boolean']>;
}>;


export type GetTasksQuery = { __typename?: 'Query', flattenedTasks: Array<{ __typename?: 'Task', name: string, id: string, effectiveDueDate?: string | null, completed: boolean, effectivelyCompleted: boolean, flagged: boolean, containingProject?: { __typename?: 'Project', id: string, name: string } | null }> };

export type GetInboxTasksQueryVariables = Exact<{
  flagged?: InputMaybe<Scalars['Boolean']>;
  available?: InputMaybe<Scalars['Boolean']>;
}>;


export type GetInboxTasksQuery = { __typename?: 'Query', inboxTasks: Array<{ __typename?: 'Task', name: string, id: string, effectiveDueDate?: string | null, completed: boolean, effectivelyCompleted: boolean, flagged: boolean, containingProject?: { __typename?: 'Project', id: string, name: string } | null }> };

export type GetTasksInProjectQueryVariables = Exact<{
  projectId: Scalars['String'];
}>;


export type GetTasksInProjectQuery = { __typename?: 'Query', project?: { __typename?: 'Project', rootTask: { __typename?: 'Task', tasks?: Array<{ __typename?: 'Task', name: string, id: string, effectiveDueDate?: string | null, completed: boolean, effectivelyCompleted: boolean, flagged: boolean, containingProject?: { __typename?: 'Project', id: string, name: string } | null }> | null } } | null };



export type ResolverTypeWrapper<T> = Promise<T> | T;


export type ResolverWithResolve<TResult, TParent, TContext, TArgs> = {
  resolve: ResolverFn<TResult, TParent, TContext, TArgs>;
};
export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> = ResolverFn<TResult, TParent, TContext, TArgs> | ResolverWithResolve<TResult, TParent, TContext, TArgs>;

export type ResolverFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => Promise<TResult> | TResult;

export type SubscriptionSubscribeFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => AsyncIterable<TResult> | Promise<AsyncIterable<TResult>>;

export type SubscriptionResolveFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => TResult | Promise<TResult>;

export interface SubscriptionSubscriberObject<TResult, TKey extends string, TParent, TContext, TArgs> {
  subscribe: SubscriptionSubscribeFn<{ [key in TKey]: TResult }, TParent, TContext, TArgs>;
  resolve?: SubscriptionResolveFn<TResult, { [key in TKey]: TResult }, TContext, TArgs>;
}

export interface SubscriptionResolverObject<TResult, TParent, TContext, TArgs> {
  subscribe: SubscriptionSubscribeFn<any, TParent, TContext, TArgs>;
  resolve: SubscriptionResolveFn<TResult, any, TContext, TArgs>;
}

export type SubscriptionObject<TResult, TKey extends string, TParent, TContext, TArgs> =
  | SubscriptionSubscriberObject<TResult, TKey, TParent, TContext, TArgs>
  | SubscriptionResolverObject<TResult, TParent, TContext, TArgs>;

export type SubscriptionResolver<TResult, TKey extends string, TParent = {}, TContext = {}, TArgs = {}> =
  | ((...args: any[]) => SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>)
  | SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>;

export type TypeResolveFn<TTypes, TParent = {}, TContext = {}> = (
  parent: TParent,
  context: TContext,
  info: GraphQLResolveInfo
) => Maybe<TTypes> | Promise<Maybe<TTypes>>;

export type IsTypeOfResolverFn<T = {}, TContext = {}> = (obj: T, context: TContext, info: GraphQLResolveInfo) => boolean | Promise<boolean>;

export type NextResolverFn<T> = () => Promise<T>;

export type DirectiveResolverFn<TResult = {}, TParent = {}, TContext = {}, TArgs = {}> = (
  next: NextResolverFn<TResult>,
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => TResult | Promise<TResult>;

/** Mapping between all available schema types and the resolvers types */
export type ResolversTypes = {
  Boolean: ResolverTypeWrapper<Scalars['Boolean']>;
  Int: ResolverTypeWrapper<Scalars['Int']>;
  Project: ResolverTypeWrapper<Project>;
  Query: ResolverTypeWrapper<{}>;
  String: ResolverTypeWrapper<Scalars['String']>;
  Task: ResolverTypeWrapper<Task>;
};

/** Mapping between all available schema types and the resolvers parents */
export type ResolversParentTypes = {
  Boolean: Scalars['Boolean'];
  Int: Scalars['Int'];
  Project: Project;
  Query: {};
  String: Scalars['String'];
  Task: Task;
};

export type ProjectResolvers<ContextType = any, ParentType extends ResolversParentTypes['Project'] = ResolversParentTypes['Project']> = {
  id?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
  name?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
  rootTask?: Resolver<ResolversTypes['Task'], ParentType, ContextType>;
  __isTypeOf?: IsTypeOfResolverFn<ParentType, ContextType>;
};

export type QueryResolvers<ContextType = any, ParentType extends ResolversParentTypes['Query'] = ResolversParentTypes['Query']> = {
  flattenedTasks?: Resolver<Array<ResolversTypes['Task']>, ParentType, ContextType, Partial<QueryFlattenedTasksArgs>>;
  inboxTasks?: Resolver<Array<ResolversTypes['Task']>, ParentType, ContextType, Partial<QueryInboxTasksArgs>>;
  project?: Resolver<Maybe<ResolversTypes['Project']>, ParentType, ContextType, RequireFields<QueryProjectArgs, 'id'>>;
};

export type TaskResolvers<ContextType = any, ParentType extends ResolversParentTypes['Task'] = ResolversParentTypes['Task']> = {
  completed?: Resolver<ResolversTypes['Boolean'], ParentType, ContextType>;
  containingProject?: Resolver<Maybe<ResolversTypes['Project']>, ParentType, ContextType>;
  effectiveDueDate?: Resolver<Maybe<ResolversTypes['String']>, ParentType, ContextType>;
  effectivelyCompleted?: Resolver<ResolversTypes['Boolean'], ParentType, ContextType>;
  flagged?: Resolver<ResolversTypes['Boolean'], ParentType, ContextType>;
  id?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
  name?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
  tasks?: Resolver<Maybe<Array<ResolversTypes['Task']>>, ParentType, ContextType>;
  __isTypeOf?: IsTypeOfResolverFn<ParentType, ContextType>;
};

export type Resolvers<ContextType = any> = {
  Project?: ProjectResolvers<ContextType>;
  Query?: QueryResolvers<ContextType>;
  Task?: TaskResolvers<ContextType>;
};


export const TaskViewModelFragmentDoc = gql`
    fragment TaskViewModel on Task {
  name
  id
  effectiveDueDate
  completed
  effectivelyCompleted
  containingProject {
    id
    name
  }
  flagged
}
    `;
export const GetTasksDocument = gql`
    query GetTasks($flagged: Boolean, $available: Boolean, $withEffectiveDueDate: Boolean) {
  flattenedTasks(
    flagged: $flagged
    available: $available
    withEffectiveDueDate: $withEffectiveDueDate
  ) {
    ...TaskViewModel
  }
}
    ${TaskViewModelFragmentDoc}`;
export const GetInboxTasksDocument = gql`
    query GetInboxTasks($flagged: Boolean, $available: Boolean) {
  inboxTasks(flagged: $flagged, available: $available) {
    ...TaskViewModel
  }
}
    ${TaskViewModelFragmentDoc}`;
export const GetTasksInProjectDocument = gql`
    query GetTasksInProject($projectId: String!) {
  project(id: $projectId) {
    rootTask {
      tasks {
        ...TaskViewModel
      }
    }
  }
}
    ${TaskViewModelFragmentDoc}`;

export type SdkFunctionWrapper = <T>(action: (requestHeaders?:Record<string, string>) => Promise<T>, operationName: string) => Promise<T>;


const defaultWrapper: SdkFunctionWrapper = (action, _operationName) => action();

export function getSdk(client: GraphQLClient, withWrapper: SdkFunctionWrapper = defaultWrapper) {
  return {
    GetTasks(variables?: GetTasksQueryVariables, requestHeaders?: Dom.RequestInit["headers"]): Promise<GetTasksQuery> {
      return withWrapper((wrappedRequestHeaders) => client.request<GetTasksQuery>(GetTasksDocument, variables, {...requestHeaders, ...wrappedRequestHeaders}), 'GetTasks');
    },
    GetInboxTasks(variables?: GetInboxTasksQueryVariables, requestHeaders?: Dom.RequestInit["headers"]): Promise<GetInboxTasksQuery> {
      return withWrapper((wrappedRequestHeaders) => client.request<GetInboxTasksQuery>(GetInboxTasksDocument, variables, {...requestHeaders, ...wrappedRequestHeaders}), 'GetInboxTasks');
    },
    GetTasksInProject(variables: GetTasksInProjectQueryVariables, requestHeaders?: Dom.RequestInit["headers"]): Promise<GetTasksInProjectQuery> {
      return withWrapper((wrappedRequestHeaders) => client.request<GetTasksInProjectQuery>(GetTasksInProjectDocument, variables, {...requestHeaders, ...wrappedRequestHeaders}), 'GetTasksInProject');
    }
  };
}
export type Sdk = ReturnType<typeof getSdk>;