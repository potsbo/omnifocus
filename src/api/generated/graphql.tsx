import { GraphQLResolveInfo } from "graphql";
import { RequestInit } from "graphql-request/dist/types.dom";
import { useQuery, UseQueryOptions } from "react-query";
import { fetch } from "../fetch";
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Project = {
  __typename?: "Project";
  id: Scalars["String"];
  name: Scalars["String"];
};

export type Query = {
  __typename?: "Query";
  flattenedTasks: Array<Maybe<Task>>;
};

export type QueryFlattenedTasksArgs = {
  available?: InputMaybe<Scalars["Boolean"]>;
  flagged?: InputMaybe<Scalars["Boolean"]>;
  limit?: InputMaybe<Scalars["Int"]>;
};

export type Task = {
  __typename?: "Task";
  completed: Scalars["Boolean"];
  containingProject?: Maybe<Project>;
  effectiveDueDate?: Maybe<Scalars["String"]>;
  effectivelyCompleted: Scalars["Boolean"];
  flagged: Scalars["Boolean"];
  id: Scalars["String"];
  name: Scalars["String"];
};

export type GetFlaggedTasksQueryVariables = Exact<{ [key: string]: never }>;

export type GetFlaggedTasksQuery = {
  __typename?: "Query";
  flattenedTasks: Array<{
    __typename?: "Task";
    name: string;
    id: string;
    effectiveDueDate?: string | null;
    completed: boolean;
    effectivelyCompleted: boolean;
    flagged: boolean;
    containingProject?: { __typename?: "Project"; id: string; name: string } | null;
  } | null>;
};

export type ResolverTypeWrapper<T> = Promise<T> | T;

export type ResolverWithResolve<TResult, TParent, TContext, TArgs> = {
  resolve: ResolverFn<TResult, TParent, TContext, TArgs>;
};
export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> =
  | ResolverFn<TResult, TParent, TContext, TArgs>
  | ResolverWithResolve<TResult, TParent, TContext, TArgs>;

export type ResolverFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => Promise<TResult> | TResult;

export type SubscriptionSubscribeFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => AsyncIterable<TResult> | Promise<AsyncIterable<TResult>>;

export type SubscriptionResolveFn<TResult, TParent, TContext, TArgs> = (
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => TResult | Promise<TResult>;

export interface SubscriptionSubscriberObject<TResult, TKey extends string, TParent, TContext, TArgs> {
  subscribe: SubscriptionSubscribeFn<{ [key in TKey]: TResult }, TParent, TContext, TArgs>;
  resolve?: SubscriptionResolveFn<TResult, { [key in TKey]: TResult }, TContext, TArgs>;
}

export interface SubscriptionResolverObject<TResult, TParent, TContext, TArgs> {
  subscribe: SubscriptionSubscribeFn<any, TParent, TContext, TArgs>;
  resolve: SubscriptionResolveFn<TResult, any, TContext, TArgs>;
}

export type SubscriptionObject<TResult, TKey extends string, TParent, TContext, TArgs> =
  | SubscriptionSubscriberObject<TResult, TKey, TParent, TContext, TArgs>
  | SubscriptionResolverObject<TResult, TParent, TContext, TArgs>;

export type SubscriptionResolver<TResult, TKey extends string, TParent = {}, TContext = {}, TArgs = {}> =
  | ((...args: any[]) => SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>)
  | SubscriptionObject<TResult, TKey, TParent, TContext, TArgs>;

export type TypeResolveFn<TTypes, TParent = {}, TContext = {}> = (
  parent: TParent,
  context: TContext,
  info: GraphQLResolveInfo
) => Maybe<TTypes> | Promise<Maybe<TTypes>>;

export type IsTypeOfResolverFn<T = {}, TContext = {}> = (
  obj: T,
  context: TContext,
  info: GraphQLResolveInfo
) => boolean | Promise<boolean>;

export type NextResolverFn<T> = () => Promise<T>;

export type DirectiveResolverFn<TResult = {}, TParent = {}, TContext = {}, TArgs = {}> = (
  next: NextResolverFn<TResult>,
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo
) => TResult | Promise<TResult>;

/** Mapping between all available schema types and the resolvers types */
export type ResolversTypes = {
  Boolean: ResolverTypeWrapper<Scalars["Boolean"]>;
  Int: ResolverTypeWrapper<Scalars["Int"]>;
  Project: ResolverTypeWrapper<Project>;
  Query: ResolverTypeWrapper<{}>;
  String: ResolverTypeWrapper<Scalars["String"]>;
  Task: ResolverTypeWrapper<Task>;
};

/** Mapping between all available schema types and the resolvers parents */
export type ResolversParentTypes = {
  Boolean: Scalars["Boolean"];
  Int: Scalars["Int"];
  Project: Project;
  Query: {};
  String: Scalars["String"];
  Task: Task;
};

export type ProjectResolvers<
  ContextType = any,
  ParentType extends ResolversParentTypes["Project"] = ResolversParentTypes["Project"]
> = {
  id?: Resolver<ResolversTypes["String"], ParentType, ContextType>;
  name?: Resolver<ResolversTypes["String"], ParentType, ContextType>;
  __isTypeOf?: IsTypeOfResolverFn<ParentType, ContextType>;
};

export type QueryResolvers<
  ContextType = any,
  ParentType extends ResolversParentTypes["Query"] = ResolversParentTypes["Query"]
> = {
  flattenedTasks?: Resolver<
    Array<Maybe<ResolversTypes["Task"]>>,
    ParentType,
    ContextType,
    Partial<QueryFlattenedTasksArgs>
  >;
};

export type TaskResolvers<
  ContextType = any,
  ParentType extends ResolversParentTypes["Task"] = ResolversParentTypes["Task"]
> = {
  completed?: Resolver<ResolversTypes["Boolean"], ParentType, ContextType>;
  containingProject?: Resolver<Maybe<ResolversTypes["Project"]>, ParentType, ContextType>;
  effectiveDueDate?: Resolver<Maybe<ResolversTypes["String"]>, ParentType, ContextType>;
  effectivelyCompleted?: Resolver<ResolversTypes["Boolean"], ParentType, ContextType>;
  flagged?: Resolver<ResolversTypes["Boolean"], ParentType, ContextType>;
  id?: Resolver<ResolversTypes["String"], ParentType, ContextType>;
  name?: Resolver<ResolversTypes["String"], ParentType, ContextType>;
  __isTypeOf?: IsTypeOfResolverFn<ParentType, ContextType>;
};

export type Resolvers<ContextType = any> = {
  Project?: ProjectResolvers<ContextType>;
  Query?: QueryResolvers<ContextType>;
  Task?: TaskResolvers<ContextType>;
};

export const GetFlaggedTasksDocument = `
    query GetFlaggedTasks {
  flattenedTasks(flagged: true) {
    name
    id
    effectiveDueDate
    completed
    effectivelyCompleted
    containingProject {
      id
      name
    }
    flagged
  }
}
    `;
export const useGetFlaggedTasksQuery = <TData = GetFlaggedTasksQuery, TError = unknown>(
  variables?: GetFlaggedTasksQueryVariables,
  options?: UseQueryOptions<GetFlaggedTasksQuery, TError, TData>
) =>
  useQuery<GetFlaggedTasksQuery, TError, TData>(
    variables === undefined ? ["GetFlaggedTasks"] : ["GetFlaggedTasks", variables],
    fetch<GetFlaggedTasksQuery, GetFlaggedTasksQueryVariables>(GetFlaggedTasksDocument, variables),
    options
  );
