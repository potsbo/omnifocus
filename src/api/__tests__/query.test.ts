import { buildExecutionContext, ExecutionContext } from "graphql/execution/execute";
import {
  GetInboxTasksDocument,
  GetTasksDocument,
  GetTasksInProjectDocument,
  GetTopLevelProjectsDocument,
} from "../generated/graphql";
import { loadSchemaSync } from "@graphql-tools/load";
import { GraphQLFileLoader } from "@graphql-tools/graphql-file-loader";
import { join } from "path";
import { genQuery } from "../query";
import prettier from "prettier";
import { GraphQLError } from "graphql";
import gql from "graphql-tag";

const schema = loadSchemaSync(join(__dirname, "..", "..", "..", "assets", "schema.graphql"), {
  loaders: [new GraphQLFileLoader()],
});

const validateExecontext = (obj: ExecutionContext | readonly GraphQLError[]): obj is ExecutionContext => {
  return "operation" in obj;
};

test("query for GetTasksDocument", () => {
  const document = GetTasksDocument;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
    variableValues: { onlyFlagged: true, onlyAvailable: true },
  }) as ExecutionContext;

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query for GetTasksDocument for forecast", () => {
  const document = GetTasksDocument;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
    variableValues: { withEffectiveDueDate: true, onlyAvailable: true },
  }) as ExecutionContext;

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query for GetInboxTasksDocument", () => {
  const document = GetInboxTasksDocument;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
  }) as ExecutionContext;

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query for GetTasksInProjectDocument", () => {
  const document = GetTasksInProjectDocument;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
    variableValues: { projectId: "foobar" },
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query for GetTopLevelProjects", () => {
  const document = GetTopLevelProjectsDocument;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
    variableValues: { projectId: "foobar" },
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query for Connection", () => {
  const document = gql`
    query Hoge {
      application {
        defaultDocument {
          projects {
            pageInfo {
              hasNextPage
            }
            edges {
              cursor
              node {
                name
              }
            }
          }
        }
      }
    }
  `;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query with inline fragment", () => {
  const document = gql`
    query {
      application {
        defaultDocument {
          folders {
            edges {
              node {
                name
                sections {
                  edges {
                    node {
                      ... on Project {
                        completed
                      }
                      ... on Folder {
                        name
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  `;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query with project interface", () => {
  const document = gql`
    query {
      application {
        defaultDocument {
          projects {
            byId(id: "hZoaSakOnG4") {
              rootTask {
                id
              }
            }
          }
        }
      }
    }
  `;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});

test("query with pageInfo", () => {
  const document = gql`
    query {
      application {
        defaultDocument {
          projects(first: 10, after: "SOME-ID") {
            pageInfo {
              hasPreviousPage
              hasNextPage
              startCursor
              endCursor
            }
          }
        }
      }
    }
  `;
  const exeContext = buildExecutionContext({
    schema: schema,
    document: document,
  });
  if (!validateExecontext(exeContext)) {
    fail();
  }

  expect(prettier.format(genQuery("OmniFocus", exeContext), { parser: "babel" })).toMatchSnapshot();
});
