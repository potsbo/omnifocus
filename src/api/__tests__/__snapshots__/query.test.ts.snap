// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query for Connection 1`] = `
"({
  projects: (() => {
    const nodes = parent.projects();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: { name: elm.name() },
        };
      }),
    };
  })(),
});
"
`;

exports[`query for GetInboxTasksDocument 1`] = `
"({
  inboxTasks: (() => {
    const nodes = parent.inboxTasks.whose({
      _and: [
        { effectivelyCompleted: { _equals: false } },
        {
          _or: [
            { effectiveDeferDate: { _equals: null } },
            { effectiveDeferDate: { _lessThan: new Date() } },
          ],
        },
      ],
    })();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: {
            name: elm.name(),
            id: elm.id(),
            effectiveDueDate: elm.effectiveDueDate(),
            completed: elm.completed(),
            effectivelyCompleted: elm.effectivelyCompleted(),
            containingProject: elm.containingProject()
              ? {
                  id: elm.containingProject().id(),
                  name: elm.containingProject().name(),
                }
              : undefined,
            flagged: elm.flagged(),
          },
        };
      }),
    };
  })(),
});
"
`;

exports[`query for GetTasksDocument 1`] = `
"const onlyFlagged = true;
const onlyAvailable = true;
const withEffectiveDueDate = false;
({
  flattenedTasks: (() => {
    const nodes = parent.flattenedTasks.whose({
      _and: [
        { effectivelyCompleted: { _equals: false } },
        { flagged: { _equals: true } },
        {
          _or: [
            { effectiveDeferDate: { _equals: null } },
            { effectiveDeferDate: { _lessThan: new Date() } },
          ],
        },
      ],
    })();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: {
            name: elm.name(),
            id: elm.id(),
            effectiveDueDate: elm.effectiveDueDate(),
            completed: elm.completed(),
            effectivelyCompleted: elm.effectivelyCompleted(),
            containingProject: elm.containingProject()
              ? {
                  id: elm.containingProject().id(),
                  name: elm.containingProject().name(),
                }
              : undefined,
            flagged: elm.flagged(),
          },
        };
      }),
    };
  })(),
});
"
`;

exports[`query for GetTasksDocument for forecast 1`] = `
"const onlyFlagged = false;
const onlyAvailable = true;
const withEffectiveDueDate = true;
({
  flattenedTasks: (() => {
    const nodes = parent.flattenedTasks.whose({
      _and: [
        { effectivelyCompleted: { _equals: false } },
        { _not: [{ effectiveDeferDate: { _equals: null } }] },
        {
          _or: [
            { effectiveDeferDate: { _equals: null } },
            { effectiveDeferDate: { _lessThan: new Date() } },
          ],
        },
      ],
    })();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: {
            name: elm.name(),
            id: elm.id(),
            effectiveDueDate: elm.effectiveDueDate(),
            completed: elm.completed(),
            effectivelyCompleted: elm.effectivelyCompleted(),
            containingProject: elm.containingProject()
              ? {
                  id: elm.containingProject().id(),
                  name: elm.containingProject().name(),
                }
              : undefined,
            flagged: elm.flagged(),
          },
        };
      }),
    };
  })(),
});
"
`;

exports[`query for GetTasksInProjectDocument 1`] = `
"const projectId = \\"foobar\\";
({
  projects: (() => {
    const nodes = parent.projects();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
        };
      }),
      byId: parent.projects.byId(projectId)
        ? {
            rootTask: {
              tasks: (() => {
                const nodes = parent.projects
                  .byId(projectId)
                  .rootTask()
                  .tasks();
                return {
                  pageInfo: {
                    hasPreviousPage: false,
                    hasNextPage: false,
                    startCursor: \\"\\",
                    endCursor: \\"\\",
                  },
                  edges: nodes.map((elm) => {
                    return {
                      cursor: elm.id(),
                      node: {
                        name: elm.name(),
                        id: elm.id(),
                        effectiveDueDate: elm.effectiveDueDate(),
                        completed: elm.completed(),
                        effectivelyCompleted: elm.effectivelyCompleted(),
                        containingProject: elm.containingProject()
                          ? {
                              id: elm.containingProject().id(),
                              name: elm.containingProject().name(),
                            }
                          : undefined,
                        flagged: elm.flagged(),
                      },
                    };
                  }),
                };
              })(),
            },
          }
        : undefined,
    };
  })(),
});
"
`;

exports[`query for GetTopLevelProjects 1`] = `
"({
  folders: (() => {
    const nodes = parent.folders();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: {
            name: elm.name(),
            id: elm.id(),
            projects: (() => {
              const nodes = elm.projects();
              return {
                pageInfo: {
                  hasPreviousPage: false,
                  hasNextPage: false,
                  startCursor: \\"\\",
                  endCursor: \\"\\",
                },
                edges: nodes.map((elm) => {
                  return {
                    cursor: elm.id(),
                    node: {
                      name: elm.name(),
                      completed: elm.completed(),
                      id: elm.id(),
                      numberOfAvailableTasks: elm.numberOfAvailableTasks(),
                    },
                  };
                }),
              };
            })(),
          },
        };
      }),
    };
  })(),
  projects: (() => {
    const nodes = parent.projects();
    return {
      pageInfo: {
        hasPreviousPage: false,
        hasNextPage: false,
        startCursor: \\"\\",
        endCursor: \\"\\",
      },
      edges: nodes.map((elm) => {
        return {
          cursor: elm.id(),
          node: {
            id: elm.id(),
            name: elm.name(),
            completed: elm.completed(),
            numberOfAvailableTasks: elm.numberOfAvailableTasks(),
          },
        };
      }),
    };
  })(),
});
"
`;
